components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.rdbms/src/main/java/org/wso2/carbon/cluster/coordinator/rdbms/RDBMSCommunicationBusContextImpl.java

```diff
@@ -730,6 +703,32 @@ public void clearMembershipEvents() throws ClusterCoordinationException {
     }
 
     /**
+     * Method to retrieve the properties map, given a ResultSet.
+     * @param resultSet the ResultSet from which the properties map needs to be retrieved
+     * @return the retrieved properties map
+     */
+    private Map<String, Object> retrievePropertiesMap(ResultSet resultSet) {
+        try {
+            if (resultSet.getBlob(3) != null) {
+                int blobLength = (int) resultSet.getBlob(3).length();
+                byte[] bytes = resultSet.getBlob(3).getBytes(1L, blobLength);
+                ByteArrayInputStream bis = new ByteArrayInputStream(bytes);
+                ObjectInputStream ois = new ObjectInputStream(bis);
+                Object blobObject = ois.readObject();
+                if (blobObject instanceof Map) {
+                    return (Map) blobObject;
+                }
+            }
+        } catch (SQLException e) {
+            String task = RDBMSConstants.TASK_GET_ALL_QUEUES;
+            throw new ClusterCoordinationException("Error occurred while " + task, e);
+        } catch (IOException | ClassNotFoundException e) {
+            throw new ClusterCoordinationException("Error retrieving the property map", e);
```

Comment:
can we give more context with the error message?

Reply:
Changes introduced with #40.